### ISSUE TITLE
`p_gai_strerror` on Windows returns error not in UTF-8

### ISSUE THREAD

**Issue by ondj** [libgit2/libgit2#4716]
### Reproduction steps

Try to clone repo on Windows from not exists domain

### Expected behavior

Error message is in UTF-8

### Actual behavior

Error message is in bad encoding (probably UTF-16?)

`failed to resolve address for not-exists-domain.test: Nen� zn�m ��dn� takov� hostitel.`  - this is how error message looks like after loosy transforming to UTF-8

### Version of libgit2 (release number or SHA1)

`771dfd1dd1c27a4693dfdfea521c07e72f456b29`

### Operating system(s) tested

Windows 10 with Czech locale

First was reported as https://github.com/alexcrichton/git2-rs/issues/324. I think that the problem is, that for getting error message is called method `p_gai_strerror` 

https://github.com/libgit2/libgit2/blob/88450c1c6043b34c4433c475dcfd4928439a4810/src/streams/socket.c#L102

that is translated to `gai_strerror` method, but the encoding is not converted like for others errors in https://github.com/libgit2/libgit2/blob/master/src/win32/error.c.

https://github.com/libgit2/libgit2/blob/ec76cc486058f36d878eca7511aaa9f764bf1b83/src/posix.h#L172

**Comment by stofte** [comment]
I was in this area the other day (also using git2 with VS2017), and I noticed that if you try to make cc-rs [compile git2 using UNICODE](https://docs.microsoft.com/en-us/windows/desktop/intl/unicode-in-the-windows-api). If I did this, curl broke on some missing symbols that were wchar to utf8 related, IIRC.

I ended up only specifying the runtime flags to avoid the compilation warnings from link.exe, but I left the other stuff as is. What you're seeing here might be because CURL is compiled in ANSI mode somehow. If you `set CLFAGS=/DUNICODE` I believe you'll see CURL wont build anymore.
